implementations/rust/ockam/ockam_transport_udp/src/rendezvous_service/messages.rs

```diff
@@ -0,0 +1,24 @@
+use std::net::SocketAddr;
+
+use ockam_core::{Message, Result};
+use serde::{Deserialize, Serialize};
+
+/// Request type for UDP Hole Punching Rendezvous service
+#[derive(Serialize, Deserialize, Debug, Message)]
+pub enum RendezvousRequest {
+    Update {
+        /// Name of sending node
+        node_name: String,
+    },
+    Query {
+        /// Name of node to lookup
+        node_name: String,
+    },
+}
+
+/// Response type for UDP Hole Punching Rendezvous service
+#[derive(Serialize, Deserialize, Debug, Message)]
+pub enum RendezvousResponse {
+    Update(Result<()>),
+    Query(Result<SocketAddr>),
+}
```

Comment:
This looks like a tiny protocol.

Leaving this comment just to note that we don't want the message serialization to be coupled with Rust's default binary encoding, so we should turn this in CBOR messages.

Reply:
@mrinalwadhwa Are you saying that this should change to use the `minicbor` encoding and `ockam_core::api::Request` like `CredentialExchangeWorker` does?

So I can learn, can I ask why would we like this to use CBOR while other messages (e.g. `PortalMessage` used by `TcpOutletListenWorker`) use Rust's default encoding?

Thanks!

Reply:
@neil2468 I think that we are supposed to move everything to minicbor now. I am going to raise this issue.

Reply:
@neil2468 I actually still need to come back to you with a better answer. The question of migrating ockam messages from BARE to CBOR [is still open](https://github.com/build-trust/ockam/issues/2496). I think that what @mrinalwadhwa meant was that for workers implementing a request/response protocol, they should follow the pattern established by workers like the `CredentialExchangeWorker` and use the `Request` API present in `ockam_core`. I will come back with a confirmation.